File: DSA/DSA06023 SẮP XẾP ĐỔI CHỖ TRỰC TIẾP.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;

    for (int i = 0; i < n-1; i++)
    {
        for (int j = i+1; j < n; j++)
        {
            if (A[i] > A[j])
            {
                swap(A[i], A[j]);
            }
        }
        cout << "Buoc " << i+1 << ": ";
        for (auto &&k : A) cout << k << " ";
        cout << endl;
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA06023 SẮP XẾP ĐỔI CHỖ TRỰC TIẾP.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))

    for i in range(n-1):
        for j in range(i+1, n):
            if A[i]>A[j]:
                A[i], A[j] = A[j], A[i]
        print(f"Buoc {i+1}:", end=" ")
        print(" ".join(map(str, A)))


submit()
